.github/workflows/main.yml

```diff
@@ -1,7 +1,4 @@
 name: 'Tests'
```

Comment:
**Suggestion**: Seeing that you added a comment about branch protection rules, here is another gotcha, that we should probably document:

This workflow name is the trigger for the create-gui-pr.yml workflow:

```
  workflow_run:
    workflows: ['Tests']
```

(Yes, it’s very unfortunate how GitHub Actions uses _job_ names, _step_ IDs, and _workflow_ names as identifying references.)

Reply:
Ah! Nice spot!

Ok umm, I was considering changing `Tests` to `tests` also so it looks like a machine identifier rather than a human arbitrary name, wdyt?

Reply:
Hm. The thing is, technically, a workflow also has an ID (I think it’s just the file name without `.yml`, e.g. `main` for this one). Maybe we can make this less painful by using a prefix convention like the following:

- Workflows are always referenced by name. Workflows must be named with a `workflow-` prefix (e.g. `workflow-tests`).
- Jobs are always referenced by ID (i.e. the name of the key under `jobs:`) were possible (e.g. in `needs: [install-dependencies]`). Were _not_ possible (e.g. in branch protection rules), jobs are referenced by name. Jobs must be named with a `job-` prefix (e.g. `job-tests`).
- Steps are always referenced by ID. Steps are not prefixed as accessing them in workflows already includes a `steps` key (e.g. `steps.node-modules-cache.outputs.cache-hit`).

Reply:
Ah, just double checking one thing, so:

https://github.com/kumahq/kuma-gui/blob/master/.github/workflows/main.yml#L1

is a completely different thing to:

https://github.com/kumahq/kuma-gui/blob/master/.github/workflows/main.yml#L48

? i.e. The strings here just happen to use the same "Tests" string, they are not really related in any way.

If thats the case, I would like to change the workflow name to something like

`name: main`

and the branch protection name to something like:

`name: main-success`

`name: tests` and `name: tests-success` would also be fine.

This way the identifiers that link things together are sufficiently generic enough not to have to keep changing them if we change the contents/purpose of the things that are linked. They also look like identifiers rather than `'` wrapped strings.

I'm not super keen on redundant prefixes, i.e. I know its a workflow as its in a workflow array and I know its a job as its in the jobs object

```
workflow_run:
    workflows: [main]
---
jobs:
  install-dependencies:
```

But all in all I/we are getting a little bike-sheddy so I'm not particular precious about any of this.